datajunction-server/datajunction_server/internal/caching/cachelib_cache.py

```diff
@@ -0,0 +1,34 @@
+"""
+Cachelib-based cache implementation
+"""
+from typing import Any, Optional
+
+from cachelib import SimpleCache
+from fastapi import Request
+
+from datajunction_server.internal.caching.interface import Cache, CacheInterface
+
+
+class CachelibCache(Cache):
+    """A standard implementation of CacheInterface that uses cachelib"""
+
+    def __init__(self):
+        super().__init__()
+        self.cache = SimpleCache()
+
+    def get(self, key: str) -> Optional[Any]:
+        """Get a cached value from the simple cache"""
+        super().get(key)
+        return self.cache.get(key)
+
+    def set(self, key: str, value: Any, timeout: int = 3600) -> None:
+        """Cache a value in the simple cache"""
+        super().set(key, value, timeout)
+        self.cache.set(key, value, timeout=timeout)
+
+
+def get_cache(request: Request) -> Optional[CacheInterface]:
```

Comment:
Why `-> Optional[CacheInterface]` instead of `-> Optional[Cache]` ?

Reply:
The main reason is that there are a number of situations where we want to disable the cache, the most obvious one is when `no-cache` is in the headers. I wanted to leave this in total control of the dependency injected function though (someone may want to extend that with very intricate logic, such as cache for requests from app A but not from app B). In order to keep that complexity out of the OSS codebase in the future, all of the code is written as "if a cache is returned, it is used", so that anyone in their own dependency can write their own rules on when to return the cache and when not to.

Wrt to OSS, not returning a cache currently only happens when `no-cache` is found in the cache headers.